src/main/scala/system/RocketChipStageGenerator.scala

```diff
@@ -3,32 +3,46 @@
 package freechips.rocketchip.system
 
 import chisel3.stage.{ChiselCli, ChiselStage}
+import firrtl.AnnotationSeq
 import firrtl.options.PhaseManager.PhaseDependency
-import firrtl.options.{Dependency, Phase, PreservesAll, Shell, StageMain}
+import firrtl.options.{Dependency, Phase, PhaseManager, PreservesAll, Shell, Stage, StageMain}
 import firrtl.stage.FirrtlCli
 import freechips.rocketchip.stage.RocketChipCli
 
-class RocketChipStage extends ChiselStage with PreservesAll[Phase] {
+/** Modified ChiselStage that includes the GenerateROMs phase */
+private[freechips] final class RocketChiselStage extends ChiselStage {
 
-  override val shell = new Shell("rocket-chip") with RocketChipCli with ChiselCli with FirrtlCli
-  override val targets: Seq[PhaseDependency] = Seq(
-    Dependency[freechips.rocketchip.stage.phases.Checks],
-    Dependency[freechips.rocketchip.stage.phases.TransformAnnotations],
-    Dependency[freechips.rocketchip.stage.phases.PreElaboration],
+  override val targets = Seq(
     Dependency[chisel3.stage.phases.Checks],
     Dependency[chisel3.stage.phases.Elaborate],
     Dependency[freechips.rocketchip.stage.phases.GenerateROMs],
     Dependency[chisel3.stage.phases.AddImplicitOutputFile],
     Dependency[chisel3.stage.phases.AddImplicitOutputAnnotationFile],
     Dependency[chisel3.stage.phases.MaybeAspectPhase],
     Dependency[chisel3.stage.phases.Emitter],
-    Dependency[chisel3.stage.phases.Convert],
+    Dependency[chisel3.stage.phases.Convert]
+  )
+
+}
+
+class RocketChipStage extends Stage with PreservesAll[Phase] {
+
+  override val shell = new Shell("rocket-chip") with RocketChipCli with ChiselCli with FirrtlCli
+  val targets: Seq[PhaseDependency] = Seq(
+    Dependency[freechips.rocketchip.stage.phases.Checks],
+    Dependency[freechips.rocketchip.stage.phases.TransformAnnotations],
+    Dependency[freechips.rocketchip.stage.phases.PreElaboration],
+    Dependency[RocketChiselStage],
     Dependency[freechips.rocketchip.stage.phases.GenerateFirrtlAnnos],
     Dependency[freechips.rocketchip.stage.phases.AddDefaultTests],
     Dependency[freechips.rocketchip.stage.phases.GenerateTestSuiteMakefrags],
-    Dependency[freechips.rocketchip.stage.phases.GenerateArtefacts],
+    Dependency[freechips.rocketchip.stage.phases.GenerateArtefacts]
   )
 
+  private val pm = new PhaseManager(targets)
+
+  override def run(annotations: AnnotationSeq): AnnotationSeq = pm.transform(annotations)
+
   // TODO: need a RunPhaseAnnotation to inject phases into ChiselStage
```

Comment:
Should this comment be taken out, or do you think this is still something to consider in the future?

Reply:
The PR can remove it, sure. TODOs are probably better tracked in issues.

Reply:
thanks!